contracts/src/v0.8/functions/dev/v1_X/accessControl/TermsOfServiceAllowList.sol

```diff
@@ -101,6 +102,34 @@ contract TermsOfServiceAllowList is ITermsOfServiceAllowList, IAccessController,
     return s_allowedSenders.values();
   }
 
+  /// @inheritdoc ITermsOfServiceAllowList
+  function getAllowedSendersCount() external view override returns (uint64) {
+    return uint64(s_allowedSenders.length());
+  }
+
+  /// @dev WARNING: getAllowedSendersInRange uses EnumerableSet .length() and .at() methods to iterate over the list
+  /// without the need for an extra mapping. These method can not guarantee the ordering when new elements are added.
+  /// Evaluate if eventual consistency will satisfy your usecase before using it.
+  function getAllowedSendersInRange(
+    uint64 allowedSenderIdStart,
+    uint64 allowedSenderIdEnd
+  ) external view override returns (address[] memory allowedSenders) {
+    if (
+      allowedSenderIdStart > allowedSenderIdEnd ||
+      allowedSenderIdEnd > s_allowedSenders.length() ||
+      s_allowedSenders.length() == 0
+    ) {
+      revert InvalidCalldata();
+    }
+
+    allowedSenders = new address[]((allowedSenderIdEnd - allowedSenderIdStart) + 1);
+    for (uint256 i = 0; i <= allowedSenderIdEnd - allowedSenderIdStart; ++i) {
+      allowedSenders[i] = s_allowedSenders.at(uint256(allowedSenderIdStart + i - 1));
```

Comment:
The arguments are meant to be 0 indexed right?

Items in list: 3
Argument to AllowedSenderStart: 0
Argument to AllowedSenderEnd: 2

On the first iteration
`allowedSenders[i] = s_allowedSenders.at(uint256(0 + 0 - 1));`